src/System.Net.HttpListener/tests/InvalidClientRequestTests.cs

```diff
@@ -0,0 +1,146 @@
+            }
+
+            yield return new object[] { "GET {path} HTTP/1.1", null, new string[] { "NoValue" }, null, "Bad Request" };
+            yield return new object[] { "GET {path} HTTP/1.1", null, new string[] { ":" }, null, "Bad Request" };
+            yield return new object[] { "GET {path} HTTP/1.1", null, new string[] { "\0:value" }, null, "Bad Request" };
+            yield return new object[] { "GET {path} HTTP/1.1", null, new string[] { "value:\0" }, null, "Bad Request" };
+
+            yield return new object[] { "GET {path} HTTP/1.1", "", null, null, "Bad Request" };
+            yield return new object[] { "GET {path} HTTP/1.1", "Host: \r\n", null, null, "Bad Request" };
+
+            // [ActiveIssue(20230, TestPlatforms.AnyUnix)]
+            if (Helpers.IsWindowsImplementation)
+            {
+                yield return new object[] { "GET /something{path} HTTP/1.1", null, null, null, "Not Found" };
+                yield return new object[] { "GET {path}../ HTTP/1.1", null, null, null, "Not Found" };
+            }
+
+            // No body
+            yield return new object[] { "POST {path} HTTP/1.1", null, null, null, "Length Required" };
+            yield return new object[] { "PUT {path} HTTP/1.1", null, null, null, "Length Required" };
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
+        public async Task GetContext_InvalidRequest_DoesNotGetContext()
+        {
+            // These tests take upwards of 20s if this uses [Theory].
+            foreach (object[] testData in InvalidRequest_TestData())
+            {
+                using (Socket client = Factory.GetConnectedSocket())
+                {
+                    string requestLine = (string)testData[0];
+                    string host = (string)testData[1];
+                    IEnumerable<string> headers = (IEnumerable<string>)testData[2];
+                    string content = (string)testData[3];
+                    string expectedMessage = (string)testData[4];
+
+                    Uri listeningUri = new Uri(Factory.ListeningUrl);
+                    string requestLineWithPathAndQuery = requestLine.Replace("{path}", listeningUri.PathAndQuery);
+
+                    host = host ?? $"Host: {listeningUri.Host}\r\n";
+                    string fullRequest = $"{requestLineWithPathAndQuery}\r\n{host}{string.Join("\r\n", headers ?? new string[0])}{content}\r\n";
+                    
+                    Task<HttpListenerContext> serverTask = Factory.GetListener().GetContextAsync();
+                    client.Send(Encoding.Default.GetBytes(fullRequest));
+
+                    byte[] errorMessageBytes = new byte[512];
+                    Task<int> clientTask = Task.Run(() => client.Receive(errorMessageBytes));
+
+                    Task completedTask = await Task.WhenAny(clientTask, serverTask);
```

Comment:
It'd probably be good to add an assert verifying that serverTask doesn't complete.